### PR TITLE
Benchmark sorting components + Tridesclous2 improvement

### PR DESCRIPTION
**Pull request by samuelgarcia**
* implement the clustering of tridesclous2 as a class
* improve benchmark components
* improve  (a bit) tridesclous2

### CODE REVIEW

File: src/spikeinterface/generation/drifting_generator.py [hunk 1]
```diff
@@ -286,6 +286,7 @@ def generate_drifting_recording(
     ),
     generate_sorting_kwargs=dict(firing_rates=(2.0, 8.0), refractory_period_ms=4.0),
     generate_noise_kwargs=dict(noise_levels=(12.0, 15.0), spatial_decay=25.0),
+    more_outputs=False,
```
Comment (alejoe91) [inline src/spikeinterface/generation/drifting_generator.py:289]:
```suggestion
    extra_outputs=False,
```

File: src/spikeinterface/sortingcomponents/benchmark/benchmark_motion_estimation.py [hunk 2]
```diff
@@ -152,23 +117,14 @@ def compute_result(self, **result_params):
...
-            gt_motion = np.mean(unit_displacements, axis=1)[:, None]
-        else:
-            # non rigid
-            gt_motion = np.zeros_like(raw_motion)
-            for t in range(temporal_bins.shape[0]):
-                f = scipy.interpolate.interp1d(
-                    self.unit_locations[:, self.direction_dim], unit_displacements[t, :], fill_value="extrapolate"
-                )
-                gt_motion[t, :] = f(spatial_bins)
+        gt_motion = get_gt_motion_from_unit_discplacement(
```
Comment (alejoe91) [inline src/spikeinterface/sortingcomponents/benchmark/benchmark_motion_estimation.py:120]:
```suggestion
        gt_motion = get_gt_motion_from_unit_displacement(
```

Comment (samuelgarcia) [reply src/spikeinterface/sortingcomponents/benchmark/benchmark_motion_estimation.py]: done

File: src/spikeinterface/generation/drifting_generator.py (hunk 1 above)
Comment (samuelgarcia) [reply src/spikeinterface/generation/drifting_generator.py]: done

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**